libs/sdk-core/src/breez_services.rs

```diff
@@ -1510,11 +1517,49 @@ impl BreezServices {
         Ok(())
     }
 
-    async fn start_signer(self: &Arc<BreezServices>, shutdown_receiver: mpsc::Receiver<()>) {
-        let signer_api = self.clone();
+    fn detect_hibernation(self: &Arc<BreezServices>) {
+        let cloned = Arc::clone(self);
+        tokio::spawn(async move {
+            loop {
+                let now = SystemTime::now();
+                tokio::time::sleep(DETECT_HIBERNATE_SLEEP_DURATION).await;
+                let elapsed = match now.elapsed() {
+                    Ok(elapsed) => elapsed,
+                    Err(e) => {
+                        error!("track_hibernation failed with: {:?}", e);
+                        continue;
+                    }
+                };
+
+                if elapsed
+                    .saturating_sub(DETECT_HIBERNATE_SLEEP_DURATION)
+                    .ge(&DETECT_HIBERNATE_MAX_OFFSET)
+                {
+                    debug!("Hibernation detected, notifying services.");
+                    let _ = cloned.hibernation_sender.send(());
```

Comment:
Perhaps we can here, just before the send, call reconnect on both breez server and geenlight?
It will save two tokio tasks that only handles these reconnections and also ensures these reconnects before the background tasks.